arcgis-ios-sdk-samples/Maps/Display layer view state/LayerStatusViewController.swift

```diff
@@ -99,44 +81,39 @@ class LayerStatusViewController: UIViewController, UITableViewDataSource, UITabl
         if status.contains(.error) {
             statuses.append("Error")
         }
+        if status.contains(.warning) {
+            statuses.append("Warning")
+        }
         if !statuses.isEmpty {
             return statuses.joined(separator: ", ")
         } else {
             return "Unknown"
         }
     }
     
-    // MARK: - Table view data source
+    // MARK: Actions
     
-    func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+    @IBAction func layerVisibilitySwitchValueChanged(_ sender: UISwitch) {
+        featureLayer.isVisible = sender.isOn
     }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return self.map?.operationalLayers.count ?? 0
-    }
+    // MARK: UIViewController
     
-    // MARK: - Table view delegates
-    
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "LayerStatusCell", for: indexPath)
-        cell.backgroundColor = .clear
-        let layer = self.map.operationalLayers[indexPath.row] as! AGSLayer
-        
-        //if the layer is loaded then show the name
-        //else use a template
-        if layer.loadStatus == .loaded {
-            cell.textLabel?.text = layer.name
-        } else {
-            cell.textLabel?.text = "Layer \(indexPath.row)"
-        }
-        
-        if let layerViewState = mapView.layerViewState(for: layer) {
-            cell.detailTextLabel?.text = viewStatusString(layerViewState.status)
-        } else {
-            cell.detailTextLabel?.text = "Unknown"
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames  = ["LayerStatusViewController"]
+        // Avoid the overlap between the status label and the map content.
+        mapView.contentInset.top = 2 * statusLabel.font.lineHeight
+        // Load the feature layer.
+        featureLayer.load { [weak self] (error) in
```

Comment:
I'll fix their orders.

In terms of the loadable pattern, I always had this question: https://github.com/Esri/arcgis-runtime-samples-ios/pull/922#discussion_r466744993

Since the feature layer is loaded from a URL, there are chances it does't load (lost connection, poor connection, etc.) I don't want to have the sample in a nothing-show-up state when the load fails, but I don't know if explicitly calling `load` makes sense. 😳

Reply:
In general, at least in the samples, we only call `load(completion:)` when an action can only be performed after something has loaded. In those cases, it is a good idea to check for an error so that we don't perform the action if the load failed. But we don't (and don't need to) explicitly call load for every map and scene. Does that make sense? In this case, nothing depends on data in the map that is only available after it is loaded, so there's no reason that I can see to explicitly call `load(completion:)` here.

Reply:
This is a nice rule of thumb! Now I have a better idea on when to use load. Thanks! 👏